src/bootstrap/src/core/build_steps/gcc.rs

```diff
@@ -112,16 +114,54 @@ impl Step for Gcc {
             return true;
         }
 
-        command(root.join("contrib/download_prerequisites")).current_dir(&root).run(builder);
-        command(root.join("configure"))
+        // GCC creates files (e.g. symlinks to the downloaded dependencies)
+        // in the source directory, which does not work with our CI setup, where we mount
+        // source directories as read-only on Linux.
+        // Therefore, as a part of the build in CI, we first copy the whole source directory
+        // to the build directory, and perform the build from there.
+        let src_dir = if CiEnv::is_ci() {
+            let src_dir = builder.gcc_out(target).join("src");
+            if src_dir.exists() {
+                builder.remove_dir(&src_dir);
+            }
+            builder.create_dir(&src_dir);
+            builder.cp_link_r(&root, &src_dir);
+            src_dir
+        } else {
+            root
+        };
+
+        command(src_dir.join("contrib/download_prerequisites")).current_dir(&src_dir).run(builder);
+        let mut configure_cmd = command(src_dir.join("configure"));
+        configure_cmd
             .current_dir(&out_dir)
+            // On CI, we compile GCC with Clang.
+            // The -Wno-everything flag is needed to make GCC compile with Clang 19.
+            // Note: this disables debug information for GCC.
+            // We might want to enable it in the future in an opt-in manner.
+            .env("CXXFLAGS", "-Wno-everything -O2")
+            .env("CFLAGS", "-Wno-everything -O2")
             .arg("--enable-host-shared")
             .arg("--enable-languages=jit")
             .arg("--enable-checking=release")
             .arg("--disable-bootstrap")
             .arg("--disable-multilib")
-            .arg(format!("--prefix={}", install_dir.display()))
-            .run(builder);
+            .arg(format!("--prefix={}", install_dir.display()));
+        let mut cc = builder.build.cc(target).display().to_string();
+        if let Some(ref ccache) = builder.build.config.ccache {
+            cc = format!("{ccache} {cc}");
+        }
```

Comment:
nit: `map_or_else` can be preferred to avoid `mut`ating `cc`.

Reply:
Hmm, does it really seem more readable/better to you?
```rust
let cc = builder
            .build
            .config
            .ccache
            .as_ref()
            .map_or_else(|| cc, |ccache| format!("{ccache} {cc}"));
```

Reply:
IMO there is no difference between these 2 in terms of readability, but one with `map_or_else` seems more convenient.